xapian-core/api/queryinternal.cc

```diff
@@ -865,94 +867,138 @@ Query::Internal::unserialise(const char ** p, const char * end,
-		}
-		case 0x0b: { // Wildcard
-		    Xapian::termcount max_expansion;
-		    if (!unpack_uint(p, end, &max_expansion) || end - *p < 2) {
-			throw SerialisationError("not enough data");
-		    }
-		    int flags = static_cast<unsigned char>(*(*p)++);
-		    op combiner = static_cast<op>(*(*p)++);
-		    string pattern;
-		    if (!unpack_string(p, end, pattern)) {
-			throw SerialisationError("not enough data");
-		    }
-		    return new Xapian::Internal::QueryWildcard(pattern,
-							       max_expansion,
-							       flags,
-							       combiner);
-		}
-		case 0x0c: { // PostingSource
-		    string name;
-		    if (!unpack_string(p, end, name)) {
-			throw SerialisationError("not enough data");
+	    switch (ch >> 4) {
+		case 1: {
+		    // 0001oxxx where:
+		    // o is the operation: LT or GT
+		    // xxx is the slot number
+		    Xapian::valueno slot = ch & 7;
+		    if (slot == 7) {
+			if (!unpack_uint(p, end, &slot)) {
+			    unpack_throw_serialisation_error(*p);
+			}
+			slot += 7;
 		    }
-
-		    const PostingSource * reg_source = reg.get_posting_source(name);
-		    if (!reg_source) {
-			string m = "PostingSource ";
-			m += name;
-			m += " not registered";
-			throw SerialisationError(m);
+		    string begin;
+		    if (!unpack_string(p, end, begin)) {
+			unpack_throw_serialisation_error(*p);
 		    }
-
-		    string serialised_source;
-		    if (!unpack_string(p, end, serialised_source)) {
-			throw SerialisationError("not enough data");
+		    switch (ch >> 3) {
+			case 2: {
+			    // 00010xxx for OP_VALUE_LT
+			    return new Xapian::Internal::QueryValueGT(slot,
+								      begin);
+			}
+			case 3: {
+			    // 00011xxx for OP_VALUE_GT
+			    string end_;
+			    if (!unpack_string(p, end, end_)) {
+				unpack_throw_serialisation_error(*p);
+			    }
+			    return new Xapian::Internal::QueryValueLT(slot,
+								      end_);
+			}
 		    }
-		    PostingSource* source =
-			reg_source->unserialise_with_registry(serialised_source,
-							      reg);
-		    return new Xapian::Internal::QueryPostingSource(source->release());
-		}
-		case 0x0d: {
-		    using Xapian::Internal::QueryScaleWeight;
-		    double scale_factor = unserialise_double(p, end);
-		    return new QueryScaleWeight(scale_factor,
-						Query(unserialise(p, end, reg)));
+		    break;
 		}
-		case 0x0e: {
-		    Xapian::termcount wqf;
-		    Xapian::termpos pos;
-		    if (!unpack_uint(p, end, &wqf) ||
-			!unpack_uint(p, end, &pos)) {
-			throw SerialisationError("not enough data");
+		case 0: {
+		    // Other operators
+		    //
+		    //   0000tttt where:
+		    //     tttt -> encodes which OP_XXX
+		    switch (ch & 0x0f) {
+			case 0x00: // OP_INVALID
+			    return new Xapian::Internal::QueryInvalid();
+			case 0x0a: { // Edit distance
+			    Xapian::termcount max_expansion;
+			    if (!unpack_uint(p, end, &max_expansion) ||
+						     end - *p < 2) {
+			    throw SerialisationError("not enough data");
+			    }
```

Comment:
There are still indentation problems such as just above - the wrapped parameter list is over-indented, while the code inside the `if` body is under-indented.